sales_team_security/security/sales_team_security.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data noupdate="0">
+
+    <record model="ir.module.category" id="module_category_sales_team">
+        <field name="name">Sales Teams</field>
+        <field name="sequence">2</field>
+    </record>
+
+    <record id="group_see_only_own_team" model="res.groups">
+        <field name="name">See only own team</field>
+        <field name="category_id" ref="sales_team_security.module_category_sales_team"/>
+    </record>
+
+</data>
+<data noupdate="1">
+
+    <record id="sale_order_team_rule" model="ir.rule">
+        <field name="name">Sales Team Orders</field>
+        <field ref="sale.model_sale_order" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="sale_order_report_team_rule" model="ir.rule">
+        <field name="name">Sales Team Orders Analysis</field>
+        <field ref="sale.model_sale_report" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="sale_order_line_team_rule" model="ir.rule">
+        <field name="name">Sales Team Order Lines</field>
+        <field ref="sale.model_sale_order_line" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
+        <field name="groups" eval="[(4, ref('sales_team_security.group_see_only_own_team'))]"/>
+    </record>
+
+    <record id="res_partner_team_rule" model="ir.rule">
+        <field name="name">Sales Team Partners</field>
+        <field ref="base.model_res_partner" name="model_id"/>
+        <field name="domain_force">['|', ('section_id', 'in', user.section_ids.ids), ('section_id', '=', False)]</field>
```

Comment:
our first solution was to write `['|',('user_id','=',user.id),('section_id.member_ids','child_of',user.id)]` but as you add section_ids to res.users this maybe can also be fast. (Our messure: 100k res.partners, 25 sales.teams, 50 res.users). Need to try out. And give Feedback.